app/src/main/java/com/shagalalab/constitution/chapter/ChapterViewModel.kt

```diff
@@ -0,0 +1,18 @@
+package com.shagalalab.constitution.chapter
+
+import androidx.lifecycle.MutableLiveData
+import androidx.lifecycle.ViewModel
+import com.shagalalab.constitution.data.dao.ChapterDao
+import com.shagalalab.constitution.data.models.ChapterModel
+import java.util.concurrent.Executors
+
+class ChapterViewModel(private val chapterDao: ChapterDao) : ViewModel() {
+
+    var chapterList: MutableLiveData<List<ChapterModel>> = MutableLiveData()
```

Comment:
I think I've mentioned this before, but without necessity you shouldn't expose `MutableLiveData` outside. In this specific case, whoever uses `chapterList` needs only to read the list, but not to write. We change the list only inside ViewModel. With its current state, anyone from outside ViewModel can change this list, because it is `MutableLiveData`, but not `LiveData`, and we don't want it. So in this kind of cases, you need to make `chapterList` private, and create another (public) variable that can be accessed from outside, and downcast it to `LiveData`. Then anyone from outside only has "read-only" access to chapterList. Do the same changes to `PartViewModel`